src/Qir/Runtime/lib/QIR/QirOutputHandling.cpp

```diff
@@ -15,9 +16,61 @@ static void PrintCStr(const char* cStr)
     WriteToCurrentStream(&msg);
 }
 
+static std::string EscapeString(std::string const& s)
+{
+    std::string out = "";
+    for (std::string::const_iterator i = s.begin(), end = s.end(); i != end; i++)
+    {
+        char c = *i;
+        if (isprint(c) && c != '\\' && c != '"')
+        {
+            out += c;
+        }
+        else
+        {
+            out += '\\';
+            switch (c)
+            {
+            case '"':
+                out += '"';
+                break;
+            case '\\':
+                out += '\\';
+                break;
+            case '\t':
+                out += 't';
+                break;
+            case '\r':
+                out += 'r';
+                break;
+            case '\n':
+                out += 'n';
+                break;
+            default:
+                char const* const hexdig = "0123456789ABCDEF";
+                out += 'x';
+                unsigned char uc = (unsigned char)c;
+                out += hexdig[uc >> 4];
+                out += hexdig[uc & 0xF];
+            }
+        }
+    }
+    return out;
+}
+
 
 extern "C"
 {
+    // Message Records
+
+    void __quantum__rt__message_record_output(QirString* qstr) // NOLINT
+    {
+        std::stringstream strStream;
+        strStream << QOH_INFO_PREFIX << EscapeString(qstr->str);
+        PrintCStr(strStream.str().c_str());
+    }
```

Comment:
What is the functionality of `EscapeString(qstr->str)` needed for? 
Why not just `strStream << QOH_INFO_PREFIX << qstr->str;`?

Reply:
I assume the reason for `EscapeString(qstr->str)` is the fact that in the fragment 
`strStream << QOH_INFO_PREFIX << EscapeString(qstr->str);`
after the part 
`strStream << QOH_INFO_PREFIX << ` 
there can be output a sequence of chars that contains the `\n`, and after that char the output will look like a next line of output (requiring its own output prefix, like `QOH_INFO_PREFIX`). 

To disambiguate that the `EscapeString(qstr->str)` is used. And later some intermediate service will revert the change of the `EscapeString(qstr->str)` and convert the output back to the one the user expects.